### ISSUE TITLE
Question about the variance of distribued training

### ISSUE THREAD

**Issue by xiezhq-hermann** [uber/horovod#332]
I deployed Horovod in my distributed training of VGG model. From the loss curve I'm quite uncertain about the training diagram, here I have some questions:

1. Are duplicated models in different GPUs have the same parameters from begining?
2. If yes, will the parameters of models diverge in the training phase for any resaon?
3. If they won't diverge, what introduce the quite different loss and validation accuracy in every ietration? 

Only the different batch of training data or some other reasons?
My assumption is the parameters of models always keep the same and the difference come from the different batch of data, since all-reduce is a synchronous training diagram either.
Am I right and could you please indicate the incorrect, thanks a lot.

**Comment by alsrgv** [comment]
@xiezhq-hermann, if you have `hvd.BroadcastGlobalVariablesHook()` or `hvd.broadcast_global_variables()` the models will all start from the same initial state, and then they should proceed with training in the lock-step.  It's common to see per-batch loss variability between batches.  The key question is whether you're seeing similar convergence as a single-GPU version in Nx fewer steps (N = total number of GPUs).
